nixos/modules/services/networking/avahi-daemon.nix

```diff
@@ -265,7 +265,7 @@ in
 
     system.nssModules = optional cfg.nssmdns pkgs.nssmdns;
     system.nssDatabases.hosts = optionals cfg.nssmdns (mkMerge [
-      (mkBefore [ "mdns_minimal [NOTFOUND=return]" ]) # before resolve
+      (mkBefore [ "mdns4_minimal [NOTFOUND=return]" ]) # before resolve
```

Comment:
This is a breaking change, and prevents hosts announcing themselves over IPv6 only (for example, because they don't ship with IPv4 or a DHCPv4 client.

This should definitely be configurable, if not opt-in.

Reply:
Linux by default tries to resolve domains via IPv6 first which causes a big problem here because all the avahi services I use (pulseaudio/pipewire zeroconf, WLED) only announces themself on IPv4. So the first resolution of something in the network takes ~30 seconds mostly because the timeout of IPv6 which makes avahi painfully slow to use and sometimes even unusable. After this change discovering things like WLED devices is near instant especially on software that run avahi in their library code like LedFX. Before this change discovering our WLED devices always took multiple tries and 5 to 10 minutes, now they are found within a few seconds.

Pipewire for example by default only publishes on IPv4 or IPv6 because otherwise the service would be duplicated (https://gitlab.freedesktop.org/prahal/pipewire/-/commit/ab5fe8957be052087178d5b96c1f31323f9360fb) 

> if not opt-in.

This wouldn't be following upstreams recommendation which in my experience is correct.

> This should definitely be configurable

How about we make this depended on the IPv6 setting already there and disable that by default?